### PR TITLE
Adding AuxRankMF test metric

### PR DESCRIPTION
**Pull request by lastmansleeping**
You can now train a SetRankEncoder model with aux loss + RankMatchFailure aux validation metrics using the below command.
```
python3 ml4ir/applications/ranking/pipeline.py \
--data_dir ml4ir/applications/ranking/tests/data/tfrecord \
--feature_config ml4ir/applications/ranking/tests/data/configs/feature_config_aux_loss.yaml \
--model_config ml4ir/applications/ranking/tests/data/configs/model_config_set_rank.yaml \
--run_id test \
--data_format tfrecord \
--execution_mode train_inference_evaluate \
--num_epochs 2 \
--loss_key softmax_cross_entropy \
--aux_loss_key aux_softmax_cross_entropy \
--aux_loss_weight 0.5
```

### CODE REVIEW

File: docs/source/misc/changelog.md [hunk 1]
```diff
@@ -5,6 +5,13 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.1.16] - 2023-02-06
+
+### Added
+- RankMatchFailure metric for evaluation
+- Statistical significance and power analysis utilities
+- Stat analysis for groupwise metrics in Ranking
+
```
Comment (lastmansleeping) [inline docs/source/misc/changelog.md:14]: @mohazahran I've updated the changelog to capture all changes we will be publishing in this new version.

File: python/ml4ir/applications/ranking/tests/test_aux_metrics_helper.py [hunk 2]
```diff
@@ -206,10 +220,73 @@ def test_compute_aux_label_metrics_invalid_click(self):
...
+
+        with self.subTest("Same ordering of aux scores with different values produce same scores"):
+            self.assertEqual(compute_rank_match_failure(aux_label_values=pd.Series([5, 0, 3, 4, 1]),
+                                                        ranks=pd.Series([1, 2, 3, 4, 5]),
+                                                        click_rank=5),
+                             compute_rank_match_failure(aux_label_values=pd.Series([10, 0, 6, 8, 2]),
+                                                        ranks=pd.Series([1, 2, 3, 4, 5]),
+                                                        click_rank=5))
+
+        with self.subTest("Dense method should be used to resolve ties in ranking aux labels"):
```
Comment (mohazahran) [inline python/ml4ir/applications/ranking/tests/test_aux_metrics_helper.py:267]: @lastmansleeping 
'Dense method" What does this mean?

Comment (lastmansleeping) [reply python/ml4ir/applications/ranking/tests/test_aux_metrics_helper.py:267]:
```
method{‘average’, ‘min’, ‘max’, ‘dense’, ‘ordinal’}, optional
The method used to assign ranks to tied elements. The following methods are available (default is ‘average’):

‘average’: The average of the ranks that would have been assigned to all the tied values is assigned to each value.

‘min’: The minimum of the ranks that would have been assigned to all the tied values is assigned to each value. (This is also referred to as “competition” ranking.)

‘max’: The maximum of the ranks that would have been assigned to all the tied values is assigned to each value.

‘dense’: Like ‘min’, but the rank of the next highest element is assigned the rank immediately after those assigned to the tied elements.

‘ordinal’: All values are given a distinct rank, corresponding to the order that the values occur in a.

```

Link -> https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.rankdata.html

**Review by mohazahran — APPROVED** [review]
LGTM
